src/environmentd/src/http/static/js/memory.jsx

```diff
@@ -27,52 +27,148 @@ async function query(sql) {
 
 const { useState, useEffect } = React;
 
-function useSQL(sql) {
-  const [response, setResponse] = useState(null);
+function ReplicaView() {
+  const [currentClusterName, setCurrentClusterName] = useState(null);
+  const [currentReplicaName, setCurrentReplicaName] = useState(null);
+  const [sqlResponse, setSqlResponse] = useState(null);
   const [loading, setLoading] = useState(true);
   const [error, setError] = useState(false);
 
+  const queryClusterReplicas = `
+    SELECT
+    clusters.name AS cluster_name, replicas.name AS replica_name
+    FROM
+      mz_catalog.mz_cluster_replicas replicas
+      LEFT JOIN mz_catalog.mz_clusters clusters ON clusters.id = replicas.cluster_id
+    ORDER BY replicas.id asc
+  `;
+
   useEffect(() => {
-    query(sql)
+    const search = new URLSearchParams(location.search);
+    const clusterName = search.get('cluster_name');
+    const replicaName = search.get('replica_name');
+    if (clusterName) {
+      setCurrentClusterName(clusterName);
+    }
+    if (replicaName) {
+      setCurrentReplicaName(replicaName);
+    }
+
+    query(queryClusterReplicas)
       .then((data) => {
-        setResponse(data);
+        const results = data.results[0].rows;
+        setSqlResponse(results);
+        if (!replicaName && results.length > 0) {
+          setCurrentClusterName(results[0][0]);
+          setCurrentReplicaName(results[0][1]);
+        }
         setLoading(false);
       })
       .catch((error) => {
         setError(error);
         setLoading(false);
       });
-  }, [sql]);
+  }, []);
 
-  return [response, loading, error];
+  useEffect(() => {
+    if (!currentReplicaName) return;
+    const params = new URLSearchParams(location.search);
+    params.set('cluster_name', currentClusterName);
+    params.set('replica_name', currentReplicaName);
+    window.history.replaceState({}, '', `${location.pathname}?${params}`);
+  }, [currentClusterName, currentReplicaName])
+
+  return (
+    <div>
+      {loading ? (
+        <div>Loading...</div>
+      ) : error ? (
+        <div>error: {error}</div>
+      ) : (
+        <div>
+          <label htmlFor="cluster_replica">Cluster Replica </label>
+          <select
+            id="cluster_replica"
+            name="cluster_replica"
+            onChange={(event) => {
+              const clusterReplica = event.target.value;
+              const clusterReplicaSplit = clusterReplica.split('|');
+              setCurrentClusterName(clusterReplicaSplit[0]);
+              setCurrentReplicaName(clusterReplicaSplit[1]);
+            }}
+            defaultValue={`${currentClusterName}|${currentReplicaName}`}>
+            {sqlResponse.map((v) => (
+              <option key={`${v[0]}|${v[1]}`} value={`${v[0]}|${v[1]}`}>
```

Comment:
Can you use `JSON.stringify` for the `value` so it's more easily parseable, and not subject to a problem if the cluster name contains a `|`?